services/src/main/java/org/keycloak/services/resources/account/AccountLoader.java

```diff
@@ -147,4 +149,21 @@ private ClientModel getAccountManagementClient(RealmModel realm) {
         return client;
     }
 
+  private AccountResourceProvider getAccountResourceProvider(Theme theme) {
+    String providerId = Constants.DEFAULT_ACCOUNT_RESOURCE_PROVIDER_ID;
+    try {
+      providerId = theme.getProperties().getProperty(Constants.ACCOUNT_RESOURCE_PROVIDER_KEY);
+    } catch (IOException e) {
+      logger.debugf(e, "Error loading theme property %s", Constants.ACCOUNT_RESOURCE_PROVIDER_KEY);
+    }
+    logger.debugf("Attempting to get AccountResourceProvider for theme %s %s", theme.getName(), providerId);
+    AccountResourceProvider provider = session.getProvider(AccountResourceProvider.class, providerId);
+    if (provider == null) {
+      // question if we should just return null, or use the default
+      return session.getProvider(AccountResourceProvider.class, Constants.DEFAULT_ACCOUNT_RESOURCE_PROVIDER_ID);
+    } else {
+      logger.debugf("Found provider %s (%s) for theme %s", providerId, provider.getClass().getName(), theme.getName());
+      return provider;
+    }
```

Comment:
Change to the following, as this uses the default provider when one hasn't been configured for the theme. This approach allows setting a different default provider for the server if needed.

```suggestion
        if (theme.getProperties().containsKey(Constants.ACCOUNT_RESOURCE_PROVIDER_KEY)) {
            return session.getProvider(AccountResourceProvider.class, theme.getProperties().getProperty(Constants.ACCOUNT_RESOURCE_PROVIDER_KEY));
        } else {
            return session.getProvider(AccountResourceProvider.class);
        }
```

Reply:
Done with try/catch for IOE